test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/ReadersTest.scala

```diff
@@ -112,13 +115,32 @@ class StringReaderTest {
   @Test def should_support_marking(): Unit = {
     assertTrue(newReader.markSupported)
   }
+
+  @Test def should_always_return_0_after_reaching_end(): Unit = {
+    val r = newReader
+    assertEquals(4, r.skip(100))
+    assertEquals(-1, r.read())
+
+    // StringReader accepts negative skip
+    assertEquals(0, r.skip(-100))
```

Comment:
Could you also add a test where the negative skip does not try to go back more than the full available length in the string? I.e., with an absolute value that is less than the string length.

For example `-2` in this case?

Reply:
Added a test dedicated for negative lookahead in https://github.com/scala-js/scala-js/pull/3665/commits/570c9f3c3a842224a26c5d1fcf8c3f9e0890e881#diff-6172b8f08f1ddaf5a345c353dbec80b0R116